src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SingleSpaceSeparatorCheck.java

```diff
@@ -170,32 +173,37 @@ public void beginTree(DetailAST rootAST) {
     }
 
     /**
-     * Examines every sibling and child of {@code node} for violations.
+     * Examines every sibling and descendant of {@code node} for violations.
      *
-     * @param node The node to start examining.
+     * @param rootAst The root of the AST
      */
-    private void visitEachToken(DetailAST node) {
-        DetailAST sibling = node;
+    private void visitEachToken(DetailAST rootAst) {
+        final Deque<DetailAST> stack = new ArrayDeque<>();
+        stack.push(rootAst);
```

Comment:
Aren't we now prone to memory overflow instead of stack?

https://github.com/checkstyle/checkstyle/issues/8943#issuecomment-721473849
> inline recursive "visit every token logic" from TreeWalker

https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java#L448-L460
Why can't we do something like this?

Reply:
I agree. I have changed the code to use the "visit every token logic" from TreeWalker.

Reply:
@rnveach Please review when you have time. Thank you